sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicy.java

```diff
@@ -67,14 +79,15 @@ public HttpPipelinePolicy create() {
             return next.process();
         }
 
+        io.opentelemetry.context.Context currentContext = io.opentelemetry.context.Context.current();
```

Comment:
Should we look into any way of passing the tracing context via the Azure Core Context using a well-known key? Basically, another way to get further away from relying on the default Thread local values.

This doesn't need to be done in this PR.

Reply:
yes, we should propagate the otel context through reactor context (instead of Span, if we can, without breaking). 
In this specific case, it's not that we even rely on the current context, we just use it to put our span there, and then we still use our span to inject context to the outgoing request. So nothing is yet broken because of it, but eventually it will be.

Reply:
https://github.com/Azure/azure-sdk-for-java

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/24281